resources/pod.py

```diff
@@ -153,3 +172,69 @@ def get_ceph_tools_pod():
     assert ct_pod, f"No Ceph tools pod found"
     ceph_pod = Pod(**ct_pod)
     return ceph_pod
+
+
+def check_file_existence(pod_obj, file_name):
+    """
+    Check if file exists inside the pod
+
+    Args:
+        pod_obj (Pod): The object of the pod
+        file_name (str): The name (full path) of the file to look for inside
+            the pod
+
+    Returns:
+        bool: True if the file exist, False otherwise
+    """
+    ret = pod_obj.exec_cmd_on_pod(f"bash -c \"find {file_name}\"")
+    if re.search(file_name, ret):
+        return True
+    return False
+
+
+def run_io_in_bg(pod_obj, expect_to_fail=False):
```

Comment:
This should be moved within the tests, we are returning a thread which can get into weird state here.
Also we have gevent based parallelism that works better, the tests should be using that instead. Ex: https://github.com/red-hat-storage/ocs-ci/blob/master/tests/ecosystem/deployment/test_ocs_basic_install.py#L201-L212

Reply:
Thanks @vasukulkarni. I will need your advice on how to use it.
I would like the command, which is infinite, to be executed on the pod, and immediately continues with the next test step, which is pvc deletion (while the I/O command is still running). I'm not sure how I can achieve that With parallel().

I tried the following which called the first function and never returned (as the run I/O command is infinite):

```python
        with parallel() as p:
            p.spawn(pod.exec_run_io_cmd, self.pod_obj, True,)
            p.spawn(self.pvc_obj.delete, False,)
```

Reply:
@ebenahar the greenlet thread will wait for join() call from each spawn'ed instance, since you are running indefinite IO, the *with* statement will be blocked here. I would suggest that we run the io here with at timeout of 600 seconds or a configurable option based on test.

Reply:
I have a few problems with using parallel in its current form:
1) I would like to wait a few seconds between I/O func call and pvc deletion 
2) I'm running infinite I/O in order to make sure that pvc deletion will take place while I/O is running and in addition, I/O should still be running while I delete the pod after pvc deletion. I want to catch the I/O command termination - CommandFailed exception.
Hence, I don't want to have a timeout.

Reply:
@ebenahar I will try to use your branch and give the diff, I think the same *sleep* you have in run_bg function should also help achieve the same thing, If not I will let you know here.

Reply:
I ran with minor changes to your code and it works with parallel, but I think I am either seeing an issue with pod exec method or rook where the IO still continues without error :( 
https://paste.fedoraproject.org/paste/s6xvyOH1RJaqGWcWst7J~A/raw

Reply:
my git diff, essential i moved while loop outside bash to see why it was still blocking https://paste.fedoraproject.org/paste/j6s59TYW4zazXuHEpWXTng/raw

Reply:
In my case, I still see this long after delete 
```
NAME                  STATUS        VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                   AGE
pvc-test-0918170846   Terminating   pvc-77a48f1d-8b1d-11e9-a9b0-061e48f3f128   1Gi        RWO            storageclass-test-0918170743   5m24s
```

Reply:
Thanks @vasukulkarni I see you are executing the oc rsh command in a while True loop. This is not good for this test as it takes a few seconds for each oc rsh command to be sent and executed. This disrupts the IO.
I think it would be better to use thread in this case.

Reply:
@ebenahar sure, I will raise a different issue for my pvc delete, also I think the granularity you are looking for can be done in threads better.